src/go/configinfo/service_info.go

```diff
@@ -317,15 +317,20 @@ func addHttpRule(method *methodInfo, r *annotationspb.HttpRule, httpMatcherWithO
 			HttpMethod:  util.PATCH,
 		}
 	case *annotationspb.HttpRule_Custom:
+		httpMethod := r.GetCustom().GetKind()
 		httpRule = &commonpb.Pattern{
 			UriTemplate: r.GetCustom().GetPath(),
-			HttpMethod:  r.GetCustom().GetKind(),
+			HttpMethod:  httpMethod,
 		}
-		matcher, err := util.GetRegexForUriWithPathParams(r.GetCustom().GetPath())
-		if err != nil {
-			matcher = r.GetCustom().GetPath()
+
+		if httpMethod == util.OPTIONS {
```

Comment:
Could you add a test for a non OPTIONS custom http rule and verify allowCors will create a rule for OPTIONS

Reply:
Done